src/V3AstNodes.cpp

```diff
@@ -61,12 +61,6 @@ void AstNodeFTaskRef::cloneRelink() {
     }
 }
 
-bool AstNodeFTaskRef::isPure() const {
-    // TODO: For non-DPI functions we could traverse the AST of function's body to determine
-    // pureness.
-    return this->taskp() && this->taskp()->dpiImport() && this->taskp()->pure();
```

Comment:
You lost check for dpiImport() I believe.

Reply:
DPI-imported functions may have side effects, but we can't check it during Verilation, so in their case we should assume that they aren't pure.

Reply:
DPI has a pure indication, we should honor that, which is why it checks taskp()->pure.

Reply:
You're right. I didn't know about the indication.